### PR TITLE
consolidate db connectors into api container

### PR DESCRIPTION
**Pull request by avimoondra**
Relates to docs overhaul of architecture concepts page - https://github.com/tryretool/docs/pull/647#discussion_r1514832936

Matches how we organize our helm deployments:

image

### CODE REVIEW

File: docker-compose-with-temporal.yml [hunk 1]
```diff
@@ -9,26 +9,21 @@ services:
...
       - WORKFLOW_BACKEND_HOST=http://workflows-backend:3000
       - WORKFLOW_TEMPORAL_CLUSTER_FRONTEND_HOST=temporal
       - WORKFLOW_TEMPORAL_CLUSTER_FRONTEND_PORT=7233
       - CODE_EXECUTOR_INGRESS_DOMAIN=http://code-executor:3004
     networks:
       - frontend-network
       - backend-network
-      - db-connector-network
-      - db-ssh-connector-network
-      - workflows-network
```
Comment (avimoondra) [inline docker-compose-with-temporal.yml:26]: renamed to temporal-network, (and renamed previous temporal-network to intra-temporal-network)

File: docker-compose-with-temporal.yml [hunk 2]
```diff
@@ -9,26 +9,21 @@ services:
...
     environment:
       - DEPLOYMENT_TEMPLATE_TYPE=docker-compose
-      - SERVICE_TYPE=MAIN_BACKEND
-      - DB_CONNECTOR_HOST=http://db-connector
-      - DB_CONNECTOR_PORT=3002
-      - DB_SSH_CONNECTOR_HOST=http://db-ssh-connector
-      - DB_SSH_CONNECTOR_PORT=3002
+      - SERVICE_TYPE=WORKFLOW_BACKEND,DB_CONNECTOR,DB_SSH_CONNECTOR
+      - DBCONNECTOR_POSTGRES_POOL_MAX_SIZE=100
+      - DBCONNECTOR_QUERY_TIMEOUT_MS=120000
```
Comment (avimoondra) [inline docker-compose-with-temporal.yml:14]: update service type

add db connector related env vars

File: docker-compose-with-temporal.yml [hunk 3]
```diff
@@ -41,6 +36,7 @@ services:
     restart: on-failure
     volumes:
       - ./keys:/root/.ssh
+      - ./keys:/retool_backend/keys
       - ssh:/retool_backend/autogen_ssh_keys
       - ./retool:/usr/local/retool-git-repo
       - ${BOOTSTRAP_SOURCE:-./retool}:/usr/local/retool-repo
```
Comment (avimoondra) [inline docker-compose-with-temporal.yml:42]: I think we can drop all these, but can do that later

(`backend/docker_scripts/generate_key_pair.sh` is deprecated)

File: docker-compose.yml
Comment (avimoondra) [inline docker-compose.yml:1]: (same changes as above)

File: docker-compose-with-temporal.yml [hunk 4]
```diff
@@ -9,26 +9,21 @@ services:
     env_file: ./docker.env
     environment:
       - DEPLOYMENT_TEMPLATE_TYPE=docker-compose
-      - SERVICE_TYPE=MAIN_BACKEND
-      - DB_CONNECTOR_HOST=http://db-connector
-      - DB_CONNECTOR_PORT=3002
-      - DB_SSH_CONNECTOR_HOST=http://db-ssh-connector
-      - DB_SSH_CONNECTOR_PORT=3002
+      - SERVICE_TYPE=WORKFLOW_BACKEND,DB_CONNECTOR,DB_SSH_CONNECTOR
```
Comment (demitrin) [inline docker-compose-with-temporal.yml:12]: probably should use `MAIN_BACKEND` here instead

**Review by demitrin — APPROVED** [review]

File: docker-compose-with-temporal.yml (hunk 4 above)
Comment (avimoondra) [reply docker-compose-with-temporal.yml]: typo, thank you!

**Review by BenjamynBaker — APPROVED** [review]

**Pull request merged**